File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/other/JigsawPuzzlePieceResponse.java
package com.kafkaquest.kq.common.util.models.events.other;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JigsawPuzzlePieceResponse {
    private UUID pieceId;
    private String pieceData;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Jigsaw/JigsawPuzzlePieceResponse.cs
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Jigsaw
{
    public class JigsawPuzzlePieceResponse
    {
        [JsonPropertyName("pieceId")]
        public Guid pieceId { get; set; }

        [JsonPropertyName("pieceData")]
        public string pieceData { get; set; }
    }
}
